akka-cluster/src/main/scala/akka/cluster/ClusterEvent.scala

```diff
@@ -457,11 +456,11 @@ object ClusterEvent {
     else {
       val otherDcs = (oldState.latestGossip.allDataCenters
           .union(newState.latestGossip.allDataCenters)) - newState.selfDc
-      otherDcs
-        .filterNot(isReachable(newState, oldState.dcReachability.allUnreachableOrTerminated))
-        .iterator
-        .map(UnreachableDataCenter)
-        .to(immutable.IndexedSeq)
+
+      val oldUnreachableDcs = otherDcs.filterNot(isDataCenterReachable(oldState))
+      val currentUnreachableDcs = otherDcs.filterNot(isDataCenterReachable(newState))
+
+      currentUnreachableDcs.diff(oldUnreachableDcs).iterator.map(UnreachableDataCenter).to(immutable.IndexedSeq)
```

Comment:
this is the same way as `diffReachableDataCenter` is implemented, but opposite

Reply:
I'm not sure why this was implemented differently. Original is from https://github.com/akka/akka/pull/23293